ethcore/src/trace/types/filter.rs

```diff
@@ -106,7 +106,7 @@ impl Filter {
 
 				let to_matches = match trace.result {
 					Res::Create(ref create_result) => self.to_address.matches(&create_result.address),
-					_ => false
+					_ => self.to_adress.matches_all(),
```

Comment:
Please add tests to cover that case and fix existing ones.

Reply:
@tomusdrw got it, add later asap.

Reply:
@tomusdrw howto add stub for created failed contract, i need some guide. could you do me a favor to give me some  smell code to reference?

Reply:
@xiaods exactly like this: https://github.com/paritytech/parity-ethereum/blob/78ba54da6bf89eff7338fe7c06d3de4d9dddeb60/ethcore/src/trace/types/filter.rs#L297

Reply:
excerpt from traceblock:
```
        {
            "action": {
                "from": "0x2a125b9db31e1257a887b674ab586c96c11e4cb7",
                "gas": "0x8af8",
                "init": "0xe20151d500000000000000000000000000000000000000000000000000000000000004570000000000000000000000000000000000000000000000000000000000000060000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000043232323200000000000000000000000000000000000000000000000000000000",
                "value": "0x0"
            },
            "blockHash": "0x667bcaac622ef255cd664fd6afc69fb86bea698de72770ea86bd71fed691a761",
            "blockNumber": 2878467,
            "error": "Bad instruction",
            "subtraces": 0,
            "traceAddress": [],
            "transactionHash": "0xb8965886e04cdbf18132a722f5a556a378b9dbcab5d1fc4050e54eadc12fafbd",
            "transactionPosition": 5,
            "type": "create"
        },
```

Reply:
Seems pretty trivial, what exactly did you try?
```
		let trace = FlatTrace {
			action: Action::Create(Create {
				from: 1.into(),
				gas: 4.into(),
				init: vec![0x5],
				value: 3.into(),
			}),
			result: Res::FailedCall(TraceError::BadInstruction),
			trace_address: vec![].into_iter().collect(),
			subtraces: 0
		};
```

The rest of the payload in RPC is just localisation data - you don't need it at all since the code you are writing just works on the traces, not their location within particular blocks.

Reply:
done. thanks for your guide.